dbt/include/athena/macros/materializations/models/helpers/get_partition_batches.sql

```diff
@@ -1,11 +1,15 @@
-{% macro get_partition_batches(sql) -%}
+{% macro get_partition_batches(sql, as_subquery=True) -%}
     {%- set partitioned_by = config.get('partitioned_by') -%}
     {%- set athena_partitions_limit = config.get('partitions_limit', 100) | int -%}
     {%- set partitioned_keys = adapter.format_partition_keys(partitioned_by) -%}
     {% do log('PARTITIONED KEYS: ' ~ partitioned_keys) %}
 
     {% call statement('get_partitions', fetch_result=True) %}
-        select distinct {{ partitioned_keys }} from ({{ sql }}) order by {{ partitioned_keys }};
+        {%- if as_subquery -%}
+            select distinct {{ partitioned_keys }} from ({{ sql }}) order by {{ partitioned_keys }};
+        {%- else -%}
+            select distinct {{ partitioned_keys }} from {{ sql }} order by {{ partitioned_keys }};
+        {%- endif -%}
```

Comment:
I'm not fully getting why we need `as_subquery`? do we have any cases where we should use as_subquery=False?
or you implemented this way to make this macro more future proof?

Reply:
yeah, here we use it https://github.com/dbt-athena/dbt-athena/pull/405/files#diff-f2eabd411bb7e43d849bdc621ef530f1304f39e18c4f1dcf73a887af4085b0d8R109

Reply:
what I mean is that we use `as_subquery=False`, means we use the tmp table that is more efficient. What will be the case where we use with `as_subquery=True`?

Reply:
ah, I just tried not to break previous logic. Mb someone uses that macro already

Reply:
ok thanks, we could leave it, as it makes it backcompatible.